pyteal/ast/ecdsa_test.py

```diff
@@ -26,22 +32,30 @@ def test_ecdsa_decompress():
         ]
     )
 
-    actual, _ = pubkey.__teal__(teal5Options)
+    actual, _ = pubkey.__teal__(curve_options_map[curve])
     actual.addIncoming()
     actual = pt.TealBlock.NormalizeBlocks(actual)
 
     with pt.TealComponent.Context.ignoreExprEquality():
         assert actual == expected
 
     # compile without errors this is necessary so assembly is also tested
-    pt.compileTeal(pt.Seq(pubkey, pt.Approve()), pt.Mode.Application, version=5)
+    pt.compileTeal(
+        pt.Seq(pubkey, pt.Approve()), pt.Mode.Application, version=curve.min_version
+    )
 
+    with pytest.raises(pt.TealInputError):
+        pt.compileTeal(
+            pt.Seq(pubkey, pt.Approve()),
+            pt.Mode.Application,
+            version=curve.min_version - 1,
+        )
 
-def test_ecdsa_recover():
+
+@pytest.mark.parametrize("curve", [pt.EcdsaCurve.Secp256k1, pt.EcdsaCurve.Secp256r1])
+def test_ecdsa_recover(curve: pt.EcdsaCurve):
```

Comment:
Recover will NOT work for Secp256r1. It would be nice to raise an error at the pyteal level for this

Reply:
@jasonpaulos I attempted to address in https://github.com/algorand/pyteal/pull/452. Can you have a look and resolve if the PR captures it?